tests/components/gogogate2/common.py

```diff
@@ -0,0 +1,162 @@
+from homeassistant.components.gogogate2.common import (
+    GogoGateDataUpdateCoordinator,
+    get_data_update_coordinator,
+)
+import homeassistant.components.gogogate2.const as const
+from homeassistant.components.homeassistant import DOMAIN as HA_DOMAIN
+from homeassistant.config import async_process_ha_core_config
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_UNIT_SYSTEM, CONF_UNIT_SYSTEM_METRIC
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+from homeassistant.setup import async_setup_component
+
+INFO_RESPONSE = InfoResponse(
+    user="user1",
+    gogogatename="gogogatename1",
+    model="",
+    apiversion="",
+    remoteaccessenabled=False,
+    remoteaccess="abcdefg.my-gogogate.com",
+    firmwareversion="",
+    apicode="API_CODE",
+    door1=Door(
+        door_id=1,
+        permission=True,
+        name="Door1",
+        mode=DoorMode.GARAGE,
+        status=DoorStatus.OPENED,
+        sensor=True,
+        sensorid=None,
+        camera=False,
+        events=2,
+        temperature=None,
+    ),
+    door2=Door(
+        door_id=2,
+        permission=True,
+        name=None,
+        mode=DoorMode.GARAGE,
+        status=DoorStatus.OPENED,
+        sensor=True,
+        sensorid=None,
+        camera=False,
+        events=2,
+        temperature=None,
+    ),
+    door3=Door(
+        door_id=3,
+        permission=True,
+        name="Door3",
+        mode=DoorMode.GARAGE,
+        status=DoorStatus.OPENED,
+        sensor=True,
+        sensorid=None,
+        camera=False,
+        events=2,
+        temperature=None,
+    ),
+    outputs=Outputs(output1=True, output2=False, output3=True),
+    network=Network(ip=""),
+    wifi=Wifi(SSID="", linkquality="", signal=""),
+)
+
+
+class ComponentData(NamedTuple):
+    """Test data for a mocked component."""
+
+    api: GogoGate2Api
+    data_update_coordinator: GogoGateDataUpdateCoordinator
+
+
+class ComponentFactory:
+    """Manages the setup and unloading of the withing component and profiles."""
+
+    def __init__(self, hass: HomeAssistant, gogogate_api_mock: Mock) -> None:
+        """Initialize the object."""
+        self._hass = hass
+        self._gogogate_api_mock = gogogate_api_mock
+
+    @property
+    def api_class_mock(self):
+        """Get the api class mock."""
+        return self._gogogate_api_mock
+
+    async def configure_component(
+        self, cover_config: Optional[List[dict]] = None
+    ) -> None:
+        """Configure the component."""
+        hass_config = {
+            "homeassistant": {CONF_UNIT_SYSTEM: CONF_UNIT_SYSTEM_METRIC},
+            "cover": cover_config or [],
+        }
+
+        await async_process_ha_core_config(self._hass, hass_config.get("homeassistant"))
+        assert await async_setup_component(self._hass, HA_DOMAIN, {})
+        assert await async_setup_component(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)